src/app/LabelLayer.tsx

```diff
@@ -0,0 +1,197 @@
+/** @jsxImportSource @emotion/react */
+
+// This file contains code related to the label step. Earlier, we had
+// code related to each step spread across multiple files, the reason
+// was that we architected the intial technical spike around tools.
+// Now, we have a single file for each step. Eventhough we have a lot
+// of repetitive code because of this change, eventually, we will
+// refactor and fix it.
+
+import { Dispatch } from "redux";
+import { LabelLayerActionMenu } from "../components/ActionMenu";
+import Annotation, {
+  AnnotationBeingCreated,
+  AnnotationProps,
+  CreationState,
+  TranslucentBox,
+} from "./Annotation";
+import { CreateAnnotationAttr, NO_OP, RenderAnnotationsHandler } from "./PDF";
+import {
+  AccessibleForm,
+  TOOL,
+  Annotation as AnnotationType,
+  Bounds,
+} from "./StoreProvider";
+
+export const labelLayerHandlers = (
+  tool: TOOL,
+  dispatch: Dispatch,
+  createAnnotationAttr: CreateAnnotationAttr
+) => {
+  const {
+    div: container,
+    creationState,
+    newCreationBounds,
+    resetCreationState,
+    updateCreationState,
+  } = createAnnotationAttr;
+
+  switch (tool) {
+    case "CREATE": {
+      return {
+        cursor: "crosshair",
+        creationState,
+        container,
+        onClick: NO_OP,
+        onMouseDown: newCreationBounds,
+        onMouseMove: updateCreationState,
+        onMouseLeave: resetCreationState,
+        onMouseUp: () => {
+          if (!creationState) return;
+          dispatch({
+            type: "CREATE_ANNOTATION_FROM_TOKENS",
+            payload: {
+              ui: {
+                id: window.crypto.randomUUID(),
+                backgroundColor: "rgb(36, 148, 178, 0.4)",
+                border: "3px solid rgb(36, 148, 178)",
+                type: "LABEL",
+              },
+              tokens: creationState.tokens,
+            },
+          });
+          resetCreationState();
+          // As soon as a label is created, we switch user to the select tool.
```

Comment:
Can we move this logic from here into the reducer, creating another action if necessary?

Reply:
Great suggestion. I am skipping this for now, but i have added a fixme for this.